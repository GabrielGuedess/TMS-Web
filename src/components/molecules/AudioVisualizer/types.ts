import { type WaveSurferOptions } from 'wavesurfer.js';

export type AudioVisualizerProps = {
  className?: string;
} & Omit<WaveSurferOptions, 'container'>;
