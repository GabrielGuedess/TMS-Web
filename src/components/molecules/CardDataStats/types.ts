import { type ElementType, type ComponentPropsWithoutRef } from 'react';

export type CardDataStatsProps = {
  rate: string;
  title: string;
  total: string;
  isPositive?: boolean;
  icon: ElementType<{ size?: number } & ComponentPropsWithoutRef<'svg'>>;
} & ComponentPropsWithoutRef<'div'>;
