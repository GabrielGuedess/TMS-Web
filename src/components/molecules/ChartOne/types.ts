import { type ComponentPropsWithoutRef } from 'react';

export type ChartOneProps = ComponentPropsWithoutRef<'div'>;

export type ChartOneStateProps = {
  series: {
    name: string;
    data: number[];
  }[];
};
