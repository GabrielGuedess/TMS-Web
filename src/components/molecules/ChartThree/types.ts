import { type ComponentPropsWithoutRef } from 'react';

export type ChartThreeProps = ComponentPropsWithoutRef<'div'>;

export type ChartThreeStateProps = {
  series: {
    name: string;
    data: number[];
  }[];
};
