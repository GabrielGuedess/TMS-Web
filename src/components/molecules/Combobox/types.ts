import { type PopoverProps } from '@radix-ui/react-popover';

export type ValueProps = {
  id: string;
  description: string;
};

export type ComboBoxProps = {
  label?: string;
  search: string;
  value: ValueProps;
  className?: string;
  isLoading?: boolean;
  placeholder: string;
  isInvalid?: boolean;
  isDisable?: boolean;
  emptyMessage: string;
  errorMessage?: string;
  values?: ValueProps[];
  placeholderCommand: string;
  setSearch: (value: string) => void;
  setValue: (value: ValueProps) => void;
} & PopoverProps;
