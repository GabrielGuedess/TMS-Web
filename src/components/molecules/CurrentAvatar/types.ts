import { type ComponentPropsWithoutRef } from 'react';

export type CurrentAvatarProps = ComponentPropsWithoutRef<'div'>;
