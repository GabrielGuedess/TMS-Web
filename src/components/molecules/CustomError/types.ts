import { type ComponentPropsWithoutRef } from 'react';

import { type ICellRendererParams } from 'ag-grid-community';

import { type OptionalProps } from 'helpers/OptionalProps';

export type CustomErrorProps = {
  noRowsMessageFunc: () => string;
} & ComponentPropsWithoutRef<'div'> &
  OptionalProps<ICellRendererParams>;
