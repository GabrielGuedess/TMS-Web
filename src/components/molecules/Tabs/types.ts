import { type Route } from 'next';
import { type LinkProps } from 'next/link';

import { type UrlObject } from 'url';

type NextRoutesProps = LinkProps<Route | UrlObject>;

export type TabProps = {
  title: string;
} & NextRoutesProps;

export type TabsProps = {
  tabs: TabProps[];
};
