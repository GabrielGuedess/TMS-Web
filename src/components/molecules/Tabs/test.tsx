import { type Route } from 'next';

import { render } from '@testing-library/react';

import { Tabs } from '.';

import { type TabProps } from './types';

jest.mock('next/navigation', () => ({
  usePathname: () => 'dashboard/groups',
}));

const tabs: TabProps[] = [
  { title: 'Geral', href: 'general' as Route },
  { title: 'Dados Pessoais', href: 'personal' as Route },
  { title: 'Permiss√µes', href: 'permissions' as Route },
  { title: 'Grupos', href: 'groups' as Route },
];

describe('<Tabs />', () => {
  it('should render the tabs correctly', () => {
    const { container } = render(<Tabs tabs={tabs} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
