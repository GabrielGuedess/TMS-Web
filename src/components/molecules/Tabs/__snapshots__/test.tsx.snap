// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render the tabs correctly 1`] = `
<div
  class="flex"
>
  <div
    class="no-scrollbar flex overflow-x-auto"
  >
    <a
      class="flex items-center justify-center gap-2 text-nowrap p-5 outline-none transition-all hover:text-primary-400 focus:text-primary-400"
      href="general"
    >
      Geral
    </a>
    <a
      class="flex items-center justify-center gap-2 text-nowrap p-5 outline-none transition-all hover:text-primary-400 focus:text-primary-400"
      href="personal"
    >
      Dados Pessoais
    </a>
    <a
      class="flex items-center justify-center gap-2 text-nowrap p-5 outline-none transition-all hover:text-primary-400 focus:text-primary-400"
      href="permissions"
    >
      Permiss√µes
    </a>
    <a
      class="flex items-center justify-center gap-2 text-nowrap p-5 outline-none transition-all text-primary-400"
      href="groups"
    >
      Grupos
    </a>
  </div>
</div>
`;
